geomstats/stratified_geometry/spider.py

```diff
@@ -0,0 +1,303 @@
+"""Class for the spider.
+
+Lead authors: Anna Calissano & Jonas Lueg
+"""
+import itertools
+
+import geomstats.backend as gs
+from geomstats.geometry.euclidean import EuclideanMetric
+from geomstats.stratified_geometry.stratified_spaces import (
+    Point,
+    PointSet,
+    PointSetGeometry,
+    _vectorize_point,
+)
+
+
+class SpiderPoint(Point):
+    r"""Class for points of the Spider.
+
+    Parameters
+    ----------
+    s : int
```

Comment:
It is recommended to avoid single-letter for variable names.

Because:
- while "s" means something to the person who wrote the code, it might not mean much to the new user. The user has to go back to the documentation each time they forget what "s" meant.
- during big refactorings of the library, "Find & Replace" variables with a single-letter name can become a nightmare.

I would suggest:
- s --> stratum
- x --> stratum_coord or coord